savant/converter/yolo.py

```diff
@@ -56,92 +104,106 @@ def __call__(
             bboxes[:, 2] -= bboxes[:, 0]
             bboxes[:, 3] -= bboxes[:, 1]
             bboxes[:, 0] += bboxes[:, 2] / 2
             bboxes[:, 1] += bboxes[:, 3] / 2
 
-        # filter by class
+        # --- class filter
         if self.class_ids:
-            class_mask = np.isin(class_ids, self.class_ids)
-            bboxes = bboxes[class_mask]
-            class_ids = class_ids[class_mask]
-            confidences = confidences[class_mask]
-
-        # filter by confidence
-        if self.confidence_threshold:
-            conf_mask = confidences > self.confidence_threshold
-            bboxes = bboxes[conf_mask]
-            class_ids = class_ids[conf_mask]
-            confidences = confidences[conf_mask]
-
-        # TODO: ability to filter by size (width, height) and aspect ratio
-        # apply class agnostic NMS (all classes are treated as one)
-        if self.nms_iou_threshold > 0 and len(confidences) > 1:
-            nms_mask = nms_cpu(bboxes, confidences, self.nms_iou_threshold, self.top_k)
-            bboxes = bboxes[nms_mask]
-            class_ids = class_ids[nms_mask]
-            confidences = confidences[nms_mask]
-
-        # select top k
-        elif len(confidences) > self.top_k:
-            top_k_mask = np.argpartition(confidences, -self.top_k)[-self.top_k :]
-            bboxes = bboxes[top_k_mask]
-            class_ids = class_ids[top_k_mask]
-            confidences = confidences[top_k_mask]
-
-        roi_left, roi_top, roi_width, roi_height = roi
-
-        # scale
-        if model.input.maintain_aspect_ratio:
-            bboxes /= min(
-                model.input.width / roi_width,
-                model.input.height / roi_height,
+            mask = np.isin(class_ids, self.class_ids)
+            bboxes, confidences, class_ids = (
+                bboxes[mask],
+                confidences[mask],
+                class_ids[mask],
+            )
+
+        # --- confidence filter
+        if self.confidence_threshold > 0:
+            mask = confidences > self.confidence_threshold
+            bboxes, confidences, class_ids = (
+                bboxes[mask],
+                confidences[mask],
+                class_ids[mask],
+            )
+
+        # --- NMS (class-agnostic)
+        if self.nms_iou_threshold > 0 and bboxes.shape[0] > 1:
+            keep = nms_cpu(bboxes, confidences, self.nms_iou_threshold, self.top_k)
+            bboxes, confidences, class_ids = (
+                bboxes[keep],
+                confidences[keep],
+                class_ids[keep],
+            )
+        elif bboxes.shape[0] > self.top_k:
+            idx = np.argpartition(confidences, -self.top_k)[-self.top_k :]
+            bboxes, confidences, class_ids = (
+                bboxes[idx],
+                confidences[idx],
+                class_ids[idx],
             )
+
+        # --- map from model input space -> ROI space
+        roi_left, roi_top, roi_w, roi_h = roi
```

Comment:
What is the reason to rename `roi_left, roi_top, roi_width, roi_height = roi` to `roi_left, roi_top, roi_w, roi_h = roi`? `roi_width` is cleared than `roi_w`.